src/components/CalendarMonth.vue

```diff
@@ -41,16 +41,29 @@ export default {
     isSelected: {
       type: Number,
       default: 0,
+    },
+    firstDayOfWeek: {
+      type: Number,
+      default: 0
     }
   },
   data() {
     return {
       daysInWeek: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
     }
   },
+  created() {
+    this.rotateDaysInWeek(this.daysInWeek, this.firstDayOfWeek);
+  },
   methods: {
     selectDate(date) {
       this.$emit('selectDate', date);
+    },
+    rotateDaysInWeek(daysInWeek, shifts) {
+      while(shifts--) {
```

Comment:
will need to handle cases when `shifts` is already a negative number

Reply:
I decided to validate it in the parent so the property injected to this child is always valid (>=0 and <= 6), hence, no need to do the same checks again. If you'd rather apply an extra check here just in case, will do!

Let me know what you think.

Reply:
I think your approach is good. Validation should be done in parent so that it wouldn't have to be done twice